src/data/equiv/mul_add.lean

```diff
@@ -465,6 +466,37 @@ lemma inv_symm : (equiv.inv G).symm = equiv.inv G := rfl
 
 end group
 
+section group_with_zero
+variables [group_with_zero G]
+
+/-- Left multiplication by a nonzero element in a `group_with_zero` is a permutation of the
+underlying type. -/
+protected def mul_left' (a : G) (ha : a ≠ 0) : perm G :=
+{ to_fun := λ x, a * x,
+  inv_fun := λ x, a⁻¹ * x,
+  left_inv := λ x, by { dsimp, rw [← mul_assoc, inv_mul_cancel ha, one_mul] },
+  right_inv := λ x, by { dsimp, rw [← mul_assoc, mul_inv_cancel ha, one_mul] } }
+
+@[simp] lemma coe_mul_left' (a : G) (ha : a ≠ 0) : ⇑(equiv.mul_left' a ha) = (*) a := rfl
+
+@[simp] lemma mul_left'_symm_apply (a : G) (ha : a ≠ 0) :
+  ((equiv.mul_left' a ha).symm : G → G) = (*) a⁻¹ := rfl
```

Comment:
Can these be generated by `@[simps]` (or `@[simps {fully_applied := ff}]` if you want equalities between functions)? 

Same for `mul_right'` and for the `homeomorph` versions.

Reply:
Oops, I pushed something to this branch when I meant to push into a new branch.

I saw some more places where `simps` can be used in this file. I'll do it myself in a later PR.

Reply:
@fpvandoorn Does this countermand your initial comment?

Reply:
Yes